docker.go

```diff
@@ -138,6 +140,16 @@ func (r DockerContainerRun) WithEntrypoint(entrypoint string) DockerContainerRun
 	return r
 }
 
+func (r DockerContainerRun) WithPublish(value string) DockerContainerRun {
+	r.publishPorts = append(r.publishPorts, value)
+	return r
+}
+
+func (r DockerContainerRun) WithPublishAll(value string) DockerContainerRun {
+	r.publishAll = true
+	return r.WithPublish(value)
+}
```

Comment:
What's the purpose of having this "pass-through" configuration?

Reply:
My initial idea was to cut down on the number of different calls a user would have to include when doing a `DockerContainerRun` execute. Rather than having to do `.WithPublish("some-port").WithPublishAll()`, the user could simplify this by doing `WithPublishAll("some-port")` to achieve the effect of running a docker command with `--publish some-port --publish-all`. I thought this would be helpful since it's a major use case in our integration tests.

However, now that you've brought my attention to it it probably makes more sense to have `WithPublishAll` take in nothing in order to mimic what that docker command actually does more closely.

Reply:
I think trying to mimic the Docker command closely is what we should shoot for here. I think part of the problem with what we had previously is that it tried to do too much magic on top of what the Docker CLI was doing.